### ISSUE TITLE
Bug: 1.16 - Unable to assign border biomes to new oceans

### ISSUE THREAD

**Issue by BiomeBundle** [PG85/OpenTerrainGenerator#615]
Adding a border to the new ocean biomes doesn't work.

**Comment by authvin** [comment]
This could be from BiomeBorderLayer#sample() calling BiomeLayers#getBiomeFromLayer(), which is checking for land bit. If it doesn't have land bit set, it will return 0 as the default ocean biome ID. That would mean BiomeBorderLayer can't see the different biomes in the ocean, just that they are oceans.

**Comment by PG85** [comment]
Correct, also a problem is that the climate-specific ocean biomes are only determined after all gen depths are processed, so only default ocean exists while processing borders/isles.

**Comment by PG85** [comment]
Since 0.0.15, multiple border biomes of the same size can now be assigned to biomes. Since all ocean types are recognised as default ocean for the purposes of border biomes, you can at least make borders for the land<>ocean boundaries, though not for the ocean<>ocean boundaries.

**Comment by PG85** [comment]
Thanks for reporting this, will use #599 for this.

**Comment by SXRWahrheit** [comment]
Implementing this requires rewriting the ocean system, targeting 1.0.0
